gfdatasource/gfdatasource

```diff
@@ -59,33 +88,136 @@ class GrafanaAPI(object):
             auth=(self.credentials.username, self.credentials.password)
         )
 
+    def update_app(self, app):
+        return requests.post(
+            '/'.join([self.base_url, 'plugins', app.id, 'settings']),
+            json=app.to_json_dict(),
+            auth=(self.credentials.username, self.credentials.password)
+        )
+
+
+def cmd_datasource(grafana_api, opts):
+    datasource_url, datasource_creds = _split_creds(opts.data_source_url)
+    datasource = DataSource(
+        name=opts.name, type=opts.type, access=opts.access,
+        url=datasource_url, credentials=datasource_creds,
+    )
+
+    def update_datasource():
+        grafana_api.update_datasource(datasource)
+
+    return update_datasource
+
+
+def cmd_app(grafana_api, opts):
+    json_data = json.loads(opts.json_data)
+    if opts.secure_json_data is not None:
+        secure_json_data = json.loads(opts.secure_json_data)
+    else:
+        secure_json_data = None
+
+    app = App(
+        id=opts.id, enabled=not opts.disabled, json_data=json_data,
+        secure_json_data=secure_json_data,
+    )
+
+    def update_app():
+        grafana_api.update_app(app)
+
+    return update_app
+
+
+class DefaultSubcommandArgParse(argparse.ArgumentParser):
+    __default_subparser = None
```

Comment:
Personally, I avoid `__foo` variables, preferring `_foo` instead, as the name-mangling only helps in tiny codebases. However, feel free to leave as is.